astropy/utils/data.py

```diff
@@ -1019,90 +1019,103 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     else:
         url_key = remote_url
 
-    try:
-        if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with _open_shelve(urlmapfn, True) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
+    url_key_list = url_key.split(',')
```

Comment:
I agree that we probably don't want to a allow a list for URLs (esp. in config files) since that is more likely to break things unexpectedly.  I did check that a URL like `http://google.com,http://duckduckgo.com` is still identified as being a URL by the `is_url` utility function.

On the overall implementation I had imagined renaming the current function into a private function `_download_file`, and then making `download_file` just do the split and iterate over the URLs calling `_download_file` with each URL and the other kwargs (and of course catching errors as needed).  I haven't thought this through carefully though, maybe there is a problem?